pkg/secrethub/client.go

```diff
@@ -263,10 +265,11 @@ func (c *Client) with(options ...ClientOption) error {
 // DefaultCredential returns a reader pointing to the configured credential,
 // sourcing it either from the SECRETHUB_CREDENTIAL environment variable or
 // from the configuration directory.
-func (c *Client) DefaultCredential() credentials.Reader {
-	envCredential := os.Getenv("SECRETHUB_CREDENTIAL")
+func (c *Client) DefaultCredential() credentials.KeyReader {
+	const credentialEnvVar = "SECRETHUB_CREDENTIAL"
+	envCredential := os.Getenv(credentialEnvVar)
 	if envCredential != "" {
-		return credentials.FromString(envCredential)
+		return credentials.FromEnv(credentialEnvVar)
```

Comment:
This change delays reading the credential from the environment variable until later when the credential is actually loaded. This technically breaks backwards compatibility and I can modify it to be (almost) the way it was and still return a nice error.

Reply:
Can you provide an example of how this would break backwards compatibility?

Reply:
Sure:
```go
os.Setenv("SECRETHUB_CREDENTIAL", "ey....") // set the env var to a valid credential
client := secrethub.NewClient()
os.Setenv("SECRETHUB_CREDENTIAL", "invalid-credential") // set the env var to an invalid credential
client.Secrets().ReadString("test/test/test") // read a valid secret
```

This used to work, as when you called `secrethub.NewClient` it took the value of the env var and stored it in the string reader so it didn't matter if you changed it later on. Now it always reads from the environment, so it matters if you change it after calling the constructor.